packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -862,10 +868,37 @@ export class ContainerRuntime
 			}
 		}
 
-		const idCompressorEnabled =
-			metadata?.idCompressorEnabled ?? runtimeOptions.enableRuntimeIdCompressor ?? false;
-		let idCompressor: (IIdCompressor & IIdCompressorCore) | undefined;
-		if (idCompressorEnabled) {
+		// Enabling the IdCompressor is a one-way operation and we only want to
+		// allow new containers to turn it on
+		let idCompressorMode: IdCompressorMode;
+		if (existing) {
+			// This setting has to be sticky for correctness:
+			// 1) if compressior is OFF, it can't be enabled, as already running clients (in given document session) do not know
+			//    how to process compressor ops
+			// 2) if it's ON, then all sessions should load compressor right away
+			// 3) Same logic applies for "delayed" mode
+			// Maybe in the future we will need to enabled (and figure how to do it safely) "delayed" -> "on" change.
+			// We could do "off" -> "on" transtition too, if all clients start loading compressor (but not using it initially) and do so for a while -
+			// this will allow clients to eventually to disregard "off" setting (when it's safe so) and start using compressor in future sessions.
+			// Everyting is possible, but it needs to be designed and executed carefully, when such need arises.
+			idCompressorMode = metadata?.idCompressorEnabled ?? "off";
+		} else {
+			// FG overwrite
+			const enabled = mc.config.getBoolean("Fluid.ContainerRuntime.IdCompressorEnabled");
```

Comment:
We should keep the runtime option to enable / disable this and use FG just to override it. Consumers would want to control it at container level and not always at host level.
From how we rolled out GC, customers wanted container level switches. It helped them to control the roll-out, tie it to specific versions, roll back versions, etc.

Reply:
As we discussed in the call, it's still there!